src/app/plan/components/Wizard/HooksFormComponent.tsx

```diff
@@ -46,6 +46,27 @@ interface IHooksFormOtherProps {
   resetAddEditState: () => void;
   currentPlan: any;
 }
+const hookNameKey = 'hookName';
+const hookImageTypeKey = 'hookImageType';
+const ansibleUploadKey = 'ansibleUpload';
+const customImageKey = 'customContainerImage';
+const ansibleRuntimeImageKey = 'ansibleRuntimeImage';
+const clusterTypeKey = 'clusterType';
+const srcServiceAccountNameKey = 'srcServiceAccountName';
+const destServiceAccountNameKey = 'destServiceAccountName';
+const srcServiceAccountNamespaceKey = 'srcServiceAccountNamespace';
+const destServiceAccountNamespaceKey = 'destServiceAccountNamespace';
+const migrationStepKey = 'migrationStep';
+
+export const HooksClusterType = {
```

Comment:
+1, I remember actual typescript enums being a PITA for some reason.

Reply:
Huh. Looks like an enum should have worked fine here, I'm curious what that PITA was.

Reply:
I think it probably would have worked better here. In my specific case, I had issues with it in the kube client where one of my goals was I wanted it all statically enforced, so for each resource I wanted them specified via enum. I might have wanted them as a static class property so you'd do something like `MigResource.MigResourceKind.MigPlan`, but I hit some weird incompatibility with it. It turned into this: https://github.com/konveyor/mig-ui/blob/master/src/client/resources/mig.ts#L19, which I wasn't entirely happy with but did the job. Now the usage is:

`const planRequest = client.create(new MigResource(MigResourceKind.MigPlan, migMeta.namespace), newMigPlan);

Example: https://github.com/konveyor/mig-ui/blob/master/src/client/resources/mig.ts#L19`